integration-tests/smoke/ccip/fee_boosting_test.go

```diff
@@ -1,121 +1,316 @@
-			sender:       common.LeftPadBytes(e.Env.Chains[chains[0]].DeployerKey.From.Bytes(), 32),
-			deployedEnv:  e,
-			onchainState: state,
-			initialPrices: changeset.InitialPrices{
-				LinkPrice: deployment.E18Mult(5),
-				WethPrice: deployment.E18Mult(9),
-				GasPrice:  changeset.ToPackedFee(big.NewInt(1.8e11), big.NewInt(0)),
-			},
-			priceFeedPrices: priceFeedPrices{
-				linkPrice: big.NewInt(4.5e18), // decrease from 5e18 to 4.5e18
-				wethPrice: big.NewInt(9e8),
-			},
-			sourceChain: chains[0],
-			destChain:   chains[1],
-		})
+	// Update token prices in destination chain FeeQuoter
+	err = updateTokensPrices(e, state, destChain, map[common.Address]*big.Int{
+		state.Chains[destChain].LinkToken.Address(): linkPrice,
+		state.Chains[destChain].Weth9.Address():     wethPrice,
 	})
-}
-
-func runFeeboostTestCase(tc feeboostTestCase) {
-	require.NoError(tc.t, changeset.AddLaneWithDefaultPricesAndFeeQuoterConfig(tc.deployedEnv.Env, tc.onchainState, tc.sourceChain, tc.destChain, false))
+	require.NoError(t, err)
 
 	startBlocks := make(map[uint64]*uint64)
 	expectedSeqNum := make(map[changeset.SourceDestPair]uint64)
 	expectedSeqNumExec := make(map[changeset.SourceDestPair][]uint64)
-	msgSentEvent := changeset.TestSendRequest(tc.t, tc.deployedEnv.Env, tc.onchainState, tc.sourceChain, tc.destChain, false, router.ClientEVM2AnyMessage{
-		Receiver:     common.LeftPadBytes(tc.onchainState.Chains[tc.destChain].Receiver.Address().Bytes(), 32),
+
+	latesthdr, err := e.Env.Chains[sourceChain].Client.HeaderByNumber(testcontext.Get(t), nil)
+	require.NoError(t, err)
+	block := latesthdr.Number.Uint64()
+	msgSentEvent := changeset.TestSendRequest(t, e.Env, state, sourceChain, destChain, false, router.ClientEVM2AnyMessage{
+		Receiver:     common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
 		Data:         []byte("message that needs fee boosting"),
 		TokenAmounts: nil,
 		FeeToken:     common.HexToAddress("0x0"),
 		ExtraArgs:    nil,
 	})
+	startBlocks[sourceChain] = &block
 	expectedSeqNum[changeset.SourceDestPair{
-		SourceChainSelector: tc.sourceChain,
-		DestChainSelector:   tc.destChain,
+		SourceChainSelector: sourceChain,
+		DestChainSelector:   destChain,
 	}] = msgSentEvent.SequenceNumber
 	expectedSeqNumExec[changeset.SourceDestPair{
-		SourceChainSelector: tc.sourceChain,
-		DestChainSelector:   tc.destChain,
+		SourceChainSelector: sourceChain,
+		DestChainSelector:   destChain,
 	}] = []uint64{msgSentEvent.SequenceNumber}
 
+	err = updateGasPrice(e, state, sourceChain, destChain, originalGasPriceDestUSD)
+	require.NoError(t, err)
+
+	// Confirm gas prices are updated
+	srcFeeQuoter := state.Chains[sourceChain].FeeQuoter
+	err = changeset.ConfirmGasPriceUpdated(t, e.Env.Chains[destChain], srcFeeQuoter, 0, originalGasPriceDestUSD)
+	require.NoError(t, err)
+
+	// Confirm that fee boosting will be triggered
+	require.True(t, willTriggerFeeBoosting(t, msgSentEvent, state, sourceChain, destChain))
```

Comment:
Can we add a comment here that this kinda thing should be supplemented by data from other sources, i.e either plugin logs ("message too costly"), OCR telemetry, or prometheus?

Reply:
Or even better, lets create a ticket for it.

Reply:
The most surefire way to confirm that it needed boosting is to see that it was in the `costlyMessages` part of the observation / outcome right?

Reply:
All messages are going to end up in the costlyMessages observation. We probably need a more fine-grained approach. Created a ticket